go-controller/pkg/ovn/controller/admin_network_policy/admin_network_policy.go

```diff
@@ -0,0 +1,681 @@
+		err = c.clearAdminNetworkPolicy(anpName)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+	// at this stage the ANP exists in the cluster
+	err = c.ensureAdminNetworkPolicy(anp)
+	if err != nil {
+		// we can ignore the error if status update doesn't succeed; best effort
+		_ = c.updateANPStatusToNotReady(anp, c.zone, err.Error())
+		if errors.Unwrap(err) != ErrorANPPriorityUnsupported && errors.Unwrap(err) != ErrorANPWithDuplicatePriority {
+			// we don't want to retry for these specific errors since they
+			// need manual intervention from users to update their CRDs
+			return nil
+		}
+		return err
+	}
+	// we can ignore the error if status update doesn't succeed; best effort
+	_ = c.updateANPStatusToReady(anp, c.zone)
+	return nil
+}
+
+// ensureAdminNetworkPolicy will handle the main reconcile logic for any given anp's
+// add/update that might be triggered either due to ANP changes or the corresponding
+// matching pod or namespace changes.
+func (c *Controller) ensureAdminNetworkPolicy(anp *anpapi.AdminNetworkPolicy) error {
+	// We only support priority ranges 0-99 in OVN-K for an ANP even if upstream
+	// supports upto 1000. The 0 (highest) corresponds to 30,000 in OVN world and
+	// 99 (lowest) corresponds to 20,100 in OVN world
+	if anp.Spec.Priority > ovnkSupportedPriorityUpperBound {
+		return fmt.Errorf("error attempting to add ANP %s with priority %d because, "+
+			"%w", anp.Name, anp.Spec.Priority, ErrorANPPriorityUnsupported)
+	}
+	desiredANPState, err := newAdminNetworkPolicyState(anp)
+	if err != nil {
+		return err
+	}
+	// At a given time only 1 ANP can exist at a given priority. If two ANPs exist with same priority
+	// the behaviour is undefined upstream but in OVNK we do not allow that
+	if existingName, loaded := c.anpPriorityMap[desiredANPState.anpPriority]; loaded && existingName != anp.Name {
+		return fmt.Errorf("error attempting to add ANP %s with priority %d when another ANP %s, "+
+			"%w", anp.Name, anp.Spec.Priority, existingName, ErrorANPWithDuplicatePriority)
+	}
+	// fetch the anpState from our cache if it exists
+	currentANPState, loaded := c.anpCache[anp.Name]
+	// Based on the latest kapi ANP, namespace and pod objects:
+	// 1) Construct Port Group name using ANP name and ports of pods in ANP subject
+	// 2) Construct Address-sets with IPs of the peers in the rules
+	// 3) Construct ACLs using AS-es and PGs
+	portGroupName, _ := getAdminNetworkPolicyPGName(desiredANPState.name, false)
+	desiredPorts, err := c.getPortsOfSubject(desiredANPState.subject)
+	if err != nil {
+		return fmt.Errorf("unable to fetch ports for anp %s: %v", desiredANPState.name, err)
+	}
+	err = c.constructIPSetOfPeers(desiredANPState)
+	if err != nil {
+		return fmt.Errorf("unable to construct IPsets for anp %s: %v", desiredANPState.name, err)
+	}
+	atLeastOneRuleUpdated := false
+	desiredACLs := c.getACLsOfRules(desiredANPState, currentANPState, portGroupName, &atLeastOneRuleUpdated, false)
```

Comment:
Why don't you just return two boolean (no need to pass a pointer I guess) for ingress and egress and then we don't have to do the potentially heavy `reflect.DeepEqual(desiredANPState.ingressRules, currentANPState.ingressRules)` (same for egress)